index.js

```diff
@@ -95,20 +147,25 @@ function testGenResponse(swagger, path, operation, response, config) {
  * @param  {string} path API path to generate tests for
  * @param  {string} operation operation of the path to generate tests for
  * @param  {json} config configuration for testGen
+ * @param  {json} parameters json for different parameters
  * @returns {string|Array} set of all tests for a path's operation
  */
-function testGenOperation(swagger, path, operation, config) {
+function testGenOperation(swagger, path, operation, config, parameters) {
   var responses = swagger.paths[path][operation].responses;
   var result = [];
   var res;
+  var origin = JSON.parse(JSON.stringify(parameters));
 
   for (res in responses) {
     if (responses.hasOwnProperty(res)) {
-      result.push(testGenResponse(swagger,
-        path,
-        operation,
-        res,
-        config));
+      parameters = JSON.parse(JSON.stringify(origin));
+      if (responses.hasOwnProperty(res)) {
+        result.push(testGenResponse(swagger,
+          path,
+          operation,
+          res,
+          config, parameters));
```

Comment:
I think you are right, I thought i have to do this is because I am processing all parameters till the response level, but actually it should be end in the operation level. I will change the structure.